packages/devtools_app/lib/src/inspector/layout_explorer/ui/widgets_theme.dart

```diff
@@ -0,0 +1,238 @@
+    final bracketIndex = widgetType.indexOf('<');
+    if (bracketIndex == -1) {
+      return widgetType;
+    }
+
+    return widgetType.substring(0, bracketIndex);
+  }
+
+  static const contentWidgetColor = Color(0xff06AC3B);
+  static const highLevelWidgetColor = Color(0xffAEAEB1);
+  static const animationWidgetColor = Color(0xffE09D0E);
+  static const otherWidgetColor = Color(0xff0EA7E0);
+
+  static const animatedTheme = WidgetTheme(
+    iconAsset: WidgetIcons.animated,
+    color: animationWidgetColor,
+  );
+
+  static const transitionTheme = WidgetTheme(
+    iconAsset: WidgetIcons.transition,
+    color: animationWidgetColor,
+  );
+
+  static const textTheme = WidgetTheme(
+    iconAsset: WidgetIcons.text,
+    color: contentWidgetColor,
+  );
+
+  static const imageTheme = WidgetTheme(
+    iconAsset: WidgetIcons.image,
+    color: contentWidgetColor,
+  );
+
+  static const tabTheme = WidgetTheme(iconAsset: WidgetIcons.tab);
+  static const scrollTheme = WidgetTheme(iconAsset: WidgetIcons.scroll);
+  static const highLevelTheme = WidgetTheme(color: highLevelWidgetColor);
+
+  static const Map<String, WidgetTheme> themeMap = {
+    // High-level
+    'RenderObjectToWidgetAdapter': WidgetTheme(
+      iconAsset: WidgetIcons.root,
+      color: highLevelWidgetColor,
+    ),
+    'CupertinoApp': highLevelTheme,
+    'MaterialApp': highLevelTheme,
+    'WidgetsApp': highLevelTheme,
+
+    // Text
+    'DefaultTextStyle': textTheme,
+    'RichText': textTheme,
+    'SelectableText': textTheme,
+    'Text': textTheme,
+
+    // Images
+    'Icon': imageTheme,
+    'Image': imageTheme,
+    'RawImage': imageTheme,
+
+    // Animations
+    'AnimatedAlign': animatedTheme,
+    'AnimatedBuilder': animatedTheme,
+    'AnimatedContainer': animatedTheme,
+    'AnimatedCrossFade': animatedTheme,
+    'AnimatedDefaultTextStyle': animatedTheme,
+    'AnimatedListState': animatedTheme,
+    'AnimatedModalBarrier': animatedTheme,
+    'AnimatedOpacity': animatedTheme,
+    'AnimatedPhysicalModel': animatedTheme,
+    'AnimatedPositioned': animatedTheme,
+    'AnimatedSize': animatedTheme,
+    'AnimatedWidget': animatedTheme,
+
+    // Transitions
+    'DecoratedBoxTransition': transitionTheme,
+    'FadeTransition': transitionTheme,
+    'PositionedTransition': transitionTheme,
+    'RotationTransition': transitionTheme,
+    'ScaleTransition': transitionTheme,
+    'SizeTransition': transitionTheme,
+    'SlideTransition': transitionTheme,
+    'Hero': WidgetTheme(
+      iconAsset: WidgetIcons.hero,
+      color: animationWidgetColor,
+    ),
+
+    // Scroll
+    'CustomScrollView': scrollTheme,
+    'DraggableScrollableSheet': scrollTheme,
+    'SingleChildScrollView': scrollTheme,
+    'Scrollable': scrollTheme,
+    'Scrollbar': scrollTheme,
+    'ScrollConfiguration': scrollTheme,
+    'GridView': WidgetTheme(iconAsset: WidgetIcons.gridView),
+    'ListView': WidgetTheme(iconAsset: WidgetIcons.listView),
```

Comment:
put this WidgetTheme into a const above since it is used in multiple places

Reply:
Done